### PR TITLE
Use .lintr to enforce tidyverse style

### PR DESCRIPTION
**Pull request by MichaelChirico**
@IndrajeetPatil is drafing a JOSS paper for {lintr} and noticed the Tidyverse style guide linting package is not yet applied to any core tidyverse packages 😅

Filing this to probe interest in setting up .lintr for core tidyverse.

There are 6 remaining wide line lints, wasn't sure whether to (1) fix them (2) set a wider default (3) `# nolint` to skip.

```
R/attach.R:1:81: style: [line_length_linter] Lines should not be more than 80 characters. This line is 101 characters.
core <- c("ggplot2", "tibble", "tidyr", "readr", "purrr", "dplyr", "stringr", "forcats", "lubridate")
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
R/conflicts.R:29:3: warning: [object_usage_linter] local variable 'tidy_names' assigned but may not be used
  tidy_names <- paste0("package:", tidyverse_packages())
  ^~~~~~~~~~
R/conflicts.R:53:81: style: [line_length_linter] Lines should not be more than 80 characters. This line is 93 characters.
  funs <- format(paste0(cli::col_blue(winner), "::", cli::col_green(paste0(names(x), "()"))))
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
R/conflicts.R:61:81: style: [line_length_linter] Lines should not be more than 80 characters. This line is 132 characters.
    cli::format_inline("Use the {.href [conflicted package](http://conflicted.r-lib.org/)} to force all conflicts to become errors")
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
R/update.R:27:81: style: [line_length_linter] Lines should not be more than 80 characters. This line is 87 characters.
  cli::cat_bullet(format(behind$package), " (", behind$local, " -> ", behind$cran, ")")
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
R/update.R:57:81: style: [line_length_linter] Lines should not be more than 80 characters. This line is 99 characters.
    paste0(cli::col_yellow(cli::style_bold(package_pad)), " (", deps$local, " < ", deps$cran, ")"),
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
```

### CODE REVIEW

File: R/attach.R [hunk 1]
```diff
@@ -1,4 +1,14 @@
...
+core <- c(
+  "ggplot2",
+  "tibble",
+  "tidyr",
+  "readr",
+  "purrr",
+  "dplyr",
+  "stringr",
+  "forcats",
+  "lubridate"
```
Comment (MichaelChirico) [inline R/attach.R:10]:
```suggestion
  "lubridate",
  NULL
```

(just something I like to do with `c()` when possible)

File: R/conflicts.R [hunk 2]
```diff
@@ -50,15 +50,19 @@ tidyverse_conflict_message <- function(x) {
...
   bullets <- paste0(
     cli::col_red(cli::symbol$cross), " ", funs, " masks ", other_calls,
     collapse = "\n"
   )
 
   conflicted <- paste0(
     cli::col_cyan(cli::symbol$info), " ",
-    cli::format_inline("Use the {.href [conflicted package](http://conflicted.r-lib.org/)} to force all conflicts to become errors")
+    "Use the ",
+    cli::format_inline("{.href [conflicted package](http://conflicted.r-lib.org/)}"),
```
Comment (MichaelChirico) [inline R/conflicts.R:64]: Just use `paste()`?

Comment (hadley) [reply R/conflicts.R:64]: I hate `paste()` with the fire of a thousand suns.

File: R/attach.R (hunk 1 above)
Comment (hadley) [reply R/attach.R:10]: I think in principle that's a good idea, but it sort of implies that this list might grow, which I think is unlikely at this point in time.

File: R/conflicts.R (hunk 2 above)
Comment (MichaelChirico) [reply R/conflicts.R:64]: that's a new one 😹

**Pull request merged**